processing/src/test/java/io/druid/segment/IndexMergerTest.java

```diff
@@ -1554,4 +1558,26 @@ private IncrementalIndex getIndexWithDims(List<String> dims)
     }
     return combiningAggregators;
   }
+
+  @Test
+  public void testDictIdSeeker() throws Exception
+  {
+    IntBuffer dimConversions = ByteBuffer.allocateDirect(3 * Ints.BYTES).asIntBuffer();
+    dimConversions.put(0);
+    dimConversions.put(2);
+    dimConversions.put(4);
+    IndexMerger.DictIdSeeker dictIdSeeker = new IndexMerger.DictIdSeeker((IntBuffer) dimConversions.asReadOnlyBuffer().rewind());
+    Assert.assertEquals(0, dictIdSeeker.seek(0));
+    Assert.assertEquals(-1, dictIdSeeker.seek(1));
+    Assert.assertEquals(1, dictIdSeeker.seek(2));
+    try {
+      dictIdSeeker.seek(1);
+    }
```

Comment:
you should add an Assert.fail(..) here or else the verification doesn't happen for the case when exception is not thrown.